src/lib/sedlexing.ml

```diff
@@ -21,24 +21,38 @@ type apos = int
 
 type lexbuf = {
   refill : Uchar.t array -> int -> int -> int;
+  bytes_per_char : Uchar.t -> int;
   mutable buf : Uchar.t array;
   mutable len : int;
-  (* Number of meaningful char in buffer *)
+  (* Number of meaningful uchar in buffer *)
   mutable offset : apos;
-  (* Position of the first char in buffer
+  (* Number of meaningful bytes in buffer *)
+  mutable bytes_offset : apos;
+  (* Position of the first uchar in buffer
       in the input stream *)
   mutable pos : int;
-  (* pos is the index in the buffer *)
+  (* Position of the first byte in buffer
+      in the input stream *)
+  mutable bytes_pos : int;
+  (* pos is the index, in uchar, in the buffer *)
   mutable curr_bol : int;
-  (* bol is the index in the input stream but not buffer *)
+  (* pos is the index, in bytes, in the buffer *)
+  mutable curr_bytes_bol : int;
+  (* bol is the index, in uchar, in the input stream but not buffer *)
   mutable curr_line : int;
-  (* start from 1, if it is 0, we would not track postion info for you *)
```

Comment:
I would keep this comment

Reply:
It should have been removed with https://github.com/ocaml-community/sedlex/commit/7581a9d8a05ca82f8d37b9f6a107ffe8e08b854b, gonna remove it now.

Reply:
I didn't realize you've completely removed the feature. You should update the doc comment of `set_position` then.

```
(** set the initial tracked input position for [lexbuf].
        If set to [Lexing.dummy_pos], Sedlexing will not track position
        information for you. *)
val set_position : lexbuf -> Lexing.position -> unit
```

Reply:
I did this:
```ocaml
(** set the initial tracked input position, in code point, for [lexbuf].
        If unspecified, byte postion is set to the same value as code
        point position. *)
 ```

I cannot see anything in the code disabling position tracking when passing `dummy_pos`, which is only defined abstractly in the [official doc](https://v2.ocaml.org/api/Lexing.html), meaning any expectation on internal values could be broken by future releases, so I just removed it.